### DISCUSSION TITLE
How to add  `--offline` to the default config

### DISCUSSION THREAD

**Discussion by halloleo** [VirtusLab/scala-cli#3023] [Using directives and cmd configuration options]
I need to run scala-cli as

    scala-cli --power --offline

because my machine is not on the internet.

The option `--power` I can add with

    $ scala-cli config power true

as a default option which is great.

I tried the same for `--offline`:

    $ scala-cli config offline true

but here I get "Error: unrecognized key offline".

How can I add  `--offline` to the default config?

Also, is there a way to suppress the informational output:

> The `--offline` option is experimental
Please bear in mind that non-ideal user experience should be expected.
If you encounter any bugs or have feedback to share, make sure to reach out to the maintenance team at https://github.com/VirtusLab/scala-cli

**Comment by Gedochao** [comment]
There's no config for offline at this time.
Let me convert this into a feature request.

As for available config keys, you can check what's available under:
```bash
scala-cli config --full-help --power
```

**Reply by Gedochao** [reply]
https://github.com/VirtusLab/scala-cli/issues/3024

**Comment by Gedochao** [comment]
BTW until a config option is added, you can rely on setting the `COURSIER_MODE` environment variable or `coursier.mode` property to `offline`.
https://scala-cli.virtuslab.org/docs/guides/power/offline#how-to-use-the-offline-mode

**Reply by halloleo** [reply]
Thanks for the info about coursier environment variable.

Just wanted to clarify, can I use the coursier variable `COURSIER_MODE` even if I do not have coursier installed? (On the air-gapped  machine I have just scala-cli.) So does scala-cli obeys `COURSIER_MODE`?

**Reply by tgodzik** [reply]
Scala CLI uses coursier API underneath for fetching dependencies, so every env variable will work.

**Reply by halloleo** [reply]
This is wonderful. It even gets rid the "_The `--offline` option is experimental_" message.
